### PR TITLE
refactor app logging

### LINKED ISSUE #4849: Log module is present but not used
Code is there but doesn't look like it's invoked from anywhere.

When fixing this, unify also with the code from #4830

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Fix #4849

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Deploy this branch on stage https://github.com/Clinical-Genomics/servers/pull/1448/files
2. Deploy on stage 
3. Check the logs created under `/tmp/tmp_scout_users_activity.log`
4. Check also the logs of the container: podman <container-id> logs

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
The cose of this PR looks quite straightforward to me, but when I try starting the app as a service on stage it starts but then the container fails immediately without printing any code or message that could be of help in the debugging. I'll try to run it as a service locally.. Perhaps I'll understand!

**Comment by northwestwitch** [comment]
> The code of this PR looks quite straightforward to me, but when I try starting the app as a service on stage it starts but then the container fails immediately without printing any code or message that could be of help in the debugging. I'll try to run it as a service locally.. Perhaps I'll understand!

Ah that's annoying: when pulling the branch from docker-hub it complains that the image is not built for architecture "arm64", of course!

**Comment by dnil** [comment]
From the log this morning it Lokes like it had trouble with the log email. Have you tried locally with email on?

**Comment by northwestwitch** [comment]
> From the log this morning it looked like it had trouble with the log email. Have you tried locally with email on?

How did you check that log, for me it was just "failed with error"

**Comment by northwestwitch** [comment]
Ah, I can ever reproduce locally. Thanks that was super helpful!

**Comment by northwestwitch** [comment]
That was definitely it! Now that's I've reintroduced the code to send the emails it works as expected!

It even writes to file in the /tmp/tmp_scout_users_activity.log file! 🥳 

Thanks @dnil !!

**Comment by northwestwitch** [comment]
> > From the log this morning it looked like it had trouble with the log email. Have you tried locally with email on?
> 
> How did you check that log, for me it was just "failed with error"

It was printing errors all along in /tmp/tmp_scout_users_activity.log. I just figured it out! 😆

**Comment by dnil** [comment]
Thank `journalctl` and the service going down repeatedly... :)

**Comment by northwestwitch** [comment]
This seems to work as intended now. Marking it as ready for review!

File: scout/server/app.py [hunk 1]
```diff
@@ -96,9 +63,7 @@ def create_app(config_file=None, config=None):
 
     app.json.sort_keys = False
 
-    current_log_level = LOG.getEffectiveLevel()
-    coloredlogs.install(level="DEBUG" if app.debug else current_log_level)
-    set_activity_log(app)
+    init_log(log=LOG, app=app)
```
Comment (dnil) [inline scout/server/app.py:66]: 👍

File: scout/log/handlers.py [hunk 2]
```diff
@@ -1,9 +1,10 @@
 # -*- coding: utf-8 -*-
-import logging
 import logging.handlers
 
 
 class TlsSMTPHandler(logging.handlers.SMTPHandler):
+    """Used by the aop to send emails."""
```
Comment (dnil) [inline scout/log/handlers.py:6]:
```suggestion
    """Used by the app to send emails."""
```

**Review by dnil — APPROVED** [review]
LGTM!

**Pull request merged**